### PR TITLE
docs: add faq docs on typescript

### PR DESCRIPTION
**Pull request by Arantiryo**

### CODE REVIEW

File: website/docs/faq/docs.js [hunk 1]
```diff
@@ -24,4 +24,8 @@ exports.docs = [
     title: 'How to fix "... is not a registered scale" error?',
     slug: '/faq/registered-scale',
   },
+  {
+    title: 'How to use react-chartjs-2 with Typescript?',
```
Comment (igorlukanin) [inline website/docs/faq/docs.js:28]:
```suggestion
    title: 'How to use react-chartjs-2 with TypeScript?',
```

File: website/docs/faq/typescript.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+slug: /faq/typescript
+---
+
+# How to use react-chartjs-2 with Typescript?
```
Comment (igorlukanin) [inline website/docs/faq/typescript.md:5]:
```suggestion
# How to use react-chartjs-2 with TypeScript?
```

File: website/docs/faq/typescript.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+`Typescript` has extremely robust type inference capabilities and most of the time
```
Comment (igorlukanin) [inline website/docs/faq/typescript.md:7]:
```suggestion
TypeScript has extremely robust type inference capabilities and most of the time
```

File: website/docs/faq/typescript.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+we can enjoy type safety and autocompletion without having to do any extra work.
+
+But occasionally types need to be set explicitly. They can be imported from `Chart.js`:
+
+```typescript
+import type { ChartData, ChartOptions } from 'chart.js';
+```
+
+and then used with `options` and `data` props.
```
Comment (igorlukanin) [inline website/docs/faq/typescript.md:16]:
```suggestion
...and then used with `options` and `data` props.
```

File: website/docs/faq/typescript.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+```typescript
+interface LineProps {
+  options: ChartOptions<'line'>;
+  data: ChartData<'line'>;
+}
+```
+
+The generic being passed is a `ChartType` that can be one of the following values:
```
Comment (igorlukanin) [inline website/docs/faq/typescript.md:25]:
```suggestion
The generic ??? being passed is a `ChartType` that can be one of the following values:
```

I think a word is missing here. Generic type?

File: website/docs/faq/typescript.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
+slug: /faq/typescript
+---
+
+# How to use react-chartjs-2 with TypeScript?
+
+`TypeScript` has extremely robust type inference capabilities and most of the time
```
Comment (dangreen) [inline website/docs/faq/typescript.md:7]:
```suggestion
TypeScript has extremely robust type inference capabilities and most of the time
```

File: website/docs/faq/typescript.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+```typescript
+interface LineProps {
+  options: ChartOptions<'line'>;
+  data: ChartData<'line'>;
+}
+```
+
+The generic type being passed is a `ChartType` that can be one of the following values:
+`bar`, `line`, `scatter`, `bubble`, `pie`, `doughnut`, `polarArea` or `radar`.
```
Comment (dangreen) [inline website/docs/faq/typescript.md:26]:
```suggestion
`'bar'`, `'line'`, `'scatter'`, `'bubble'`, `'pie'`, `'doughnut'`, `'polarArea'` or `'radar'`.
```

**Review by dangreen — CHANGES_REQUESTED** [review]

**Review by dangreen — APPROVED** [review]

**Pull request merged**